components/org.wso2.carbon.identity.application.authenticator.facebook/src/main/java/org/wso2/carbon/identity/application/authenticator/facebook/FacebookAuthenticator.java

```diff
@@ -415,6 +424,30 @@ protected String getLoginType(HttpServletRequest request) {
     }
 
     @Override
+    public String getClaimDialectURI() {
+        return FacebookAuthenticatorConstants.CLAIM_DIALECT_URI;
+    }
+
+    /**
+     * This method is to associate the specified value with the specified key in MAP.
+     *
+     * @param claimUri The Claim URI
+     * @param claims   The map
+     * @param value    The value needs to be added in the MAP
+     */
+    private void generateClaims(String claimUri, Map<ClaimMapping, String> claims, String value) {
```

Comment:
Change method to build claim mapping. Append the dialect at this point and make this method protected.